text/000-rustdoc-bundle-local-resources.md

```diff
@@ -0,0 +1,106 @@
+
+The impact on >docs.rs> would also be very minimal as the size of a published crate resources is limited to a few megabytes. The only thing needed would be to handle the new `doc.files` folder.
+
+There are two options that will be impacted by this RFC: the favicon and logo that you can respectively set through:
+
+```rust
+#![doc(html_favicon_url = "some_path", html_logo_url = "some_other_path")]
+```
+
+They will follow the same rule as for other images: if this is a local path, the local file will be copied and the paths to it will be rewritten.
+
+To support `#[doc(inline)]` for foreign items using local resources, it will rely on the `-Zrustdoc-map` option.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A new rustdoc pass will be added which would go through all documentation to gather local resources into a map.
+
+Then in HTML documentation generation, the local resources pathes will be replaced by their equivalent linking to the output directory instead.
+
+The local resources files will be renamed as follows: `{original filename}-{hash}{extension}`. The `{hash}` information will be computed from the local resource file content.
+
+You can look at what the implementation could look like in [#107640](https://github.com/rust-lang/rust/pull/107640).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Allowing local resources in rustdoc output could lead to big output files if users include big resource files. This could lead to slower build times and increase the size of generated documentation (in particular in case of very big local resources!).
+
+Another problem is that people will add images into their published crates, increasing the package size whereas it's only used for documentation.
+
+# Prior art
+[prior-art]: #prior-art
+
+- [sphinx](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-latex_additional_files)
+- [haddock](https://haskell-haddock.readthedocs.io/en/latest/invoking.html?highlight=image#cmdoption-theme): it's mentioned in this command documentation that local files in the given directory will be copied into the generated output directory.
+- [doxygen](https://doxygen.nl/manual/commands.html#cmdimage): supported through `\image`.
+
+Another approach to this feature:
+
+[ePUB packages](https://www.w3.org/publishing/epub3/epub-packages.html#sec-pkg-manifest) use explicitly-declared manifests. It allows to have a fallback chain mechanism (going through resources for an entry until an available resource is found).
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Currently, to provide resources, users need to specify external URLs for resources or inline them (if possible like the `svg` image format) directly into the documentation. It has the advantage to avoid the problem of large output files, but it also requires users to upload their resources to a web server to make them available everywhere.
+
```

Comment:
It might be worth switching this to talk about `--extern-html-root-url` here, since that's the flag for `rustdoc`. `-Zrustdoc-map` is a flag to tell `cargo` to automatically generate the `--extern-html-root-url` flags based on the dependencies.